engine/common/rpc/convert/convert.go

```diff
@@ -125,50 +123,261 @@ func BlockHeaderToMessage(h *flow.Header) (*entities.BlockHeader, error) {
 
 	t := timestamppb.New(h.Timestamp)
 
+	parentVoterIds := IdentifiersToMessages(h.ParentVoterIDs)
+
 	return &entities.BlockHeader{
-		Id:        id[:],
-		ParentId:  h.ParentID[:],
-		Height:    h.Height,
-		Timestamp: t,
+		Id:                 id[:],
+		ParentId:           h.ParentID[:],
+		Height:             h.Height,
+		PayloadHash:        h.PayloadHash[:],
+		Timestamp:          t,
+		View:               h.View,
+		ParentVoterIds:     parentVoterIds,
+		ParentVoterSigData: h.ParentVoterSigData,
+		ProposerId:         h.ProposerID[:],
+		ProposerSigData:    h.ProposerSigData,
+		ChainId:            h.ChainID.String(),
 	}, nil
 }
 
+func MessageToBlockHeader(m *entities.BlockHeader) (*flow.Header, error) {
+	parentVoterIds := MessagesToIdentifiers(m.ParentVoterIds)
+
+	return &flow.Header{
+		ParentID:           MessageToIdentifier(m.ParentId),
+		Height:             m.Height,
+		PayloadHash:        MessageToIdentifier(m.PayloadHash),
+		Timestamp:          m.Timestamp.AsTime(),
+		View:               m.View,
+		ParentVoterIDs:     parentVoterIds,
+		ParentVoterSigData: m.ParentVoterSigData,
+		ProposerID:         MessageToIdentifier(m.ProposerId),
+		ProposerSigData:    m.ProposerSigData,
+		ChainID:            flow.ChainID(m.ChainId),
```

Comment:
Added a function to guard and throw and error for this, great catch! also the `TestGetBlockByIDAndHeight` function already line covers the code I've added with fixture inputs, since all these converters are just for nested structs in `block.go`